src/include/network/routines.rb

```diff
@@ -300,88 +352,42 @@ def UsedNicName(name)
       Builtins.contains(NetworkInterfaces.List(""), name)
     end
 
+    CONTROLLERS.map do |id, name|
+      define_method "#{name}_controller?" do |class_id|
+        class_id == id
+      end
+
+      define_method "#{name}_subclass" do |hwdevice|
+        subclass_id = hwdevice["sub_class_id"]
+        ret = SUBCLASSES[name][subclass_id]
+        unless ret
+          log.error "Unknown network controller type: #{hwdevice}"
+          log.error "It's probably missing in hwinfo (NOT src/hd/hd.h:sc_net_if)" if subclass_id == 128
+        end
+        ret || ""
+      end
+    end
+
     # Simple convertor from subclass to controller type.
     # @param [Hash] hwdevice map with card info containing "subclass"
     # @return short device name
     # @example ControllerType(<ethernet controller map>) -> "eth"
     def ControllerType(hwdevice)
-      hwdevice = deep_copy(hwdevice)
-      return "modem" if Ops.get_string(hwdevice, "subclass", "") == "Modem"
-      return "isdn" if Ops.get_string(hwdevice, "subclass", "") == "ISDN"
-      return "dsl" if Ops.get_string(hwdevice, "subclass", "") == "DSL"
+      class_id    = hwdevice["class_id"]
+      subclass_id = hwdevice["sub_class_id"]
 
-      subclass_id = Ops.get_integer(hwdevice, "sub_class_id", -1)
+      return "modem" if hwdevice["subclass"] == "Modem" || modem_controller?(class_id)
+      return "isdn"  if hwdevice["subclass"] == "ISDN" || isdn_controller?(class_id)
+      return "dsl"   if hwdevice["subclass"] == "DSL" || dsl_controller?(class_id)
```

Comment:
So if it was dropped i could drop all references for it and for example only support ReadHardware("netcard")